src/events.ts

```diff
@@ -83,4 +86,7 @@ export class OrbEmitter<N extends INodeBase, E extends IEdgeBase> extends Emitte
   [OrbEventType.NODE_DRAG_START]: IOrbEventMouseNodeEvent<N, E> & IOrbEventMouseMoveEvent;
   [OrbEventType.NODE_DRAG]: IOrbEventMouseNodeEvent<N, E> & IOrbEventMouseMoveEvent;
   [OrbEventType.NODE_DRAG_END]: IOrbEventMouseNodeEvent<N, E> & IOrbEventMouseMoveEvent;
+  [OrbEventType.NODE_RIGHT_CLICK]: IOrbEventMouseNodeEvent<N, E> & IOrbEventMouseClickEvent;
+  [OrbEventType.EDGE_RIGHT_CLICK]: IOrbEventMouseEdgeEvent<N, E> & IOrbEventMouseClickEvent;
+  [OrbEventType.CANVAS_RIGHT_CLICK]: IOrbEventMouseClickEvent;
```

Comment:
```suggestion
  [OrbEventType.MOUSE_RIGHT_CLICK]: IOrbEventMouseEvent<N, E> & IOrbEventMouseClickEvent;
```
I would suggest having a bit more generic event such as `mouse-right-click` that can handle any of the following three scenarios:

* Node is right-clicked, `subject` will be `INode`
* Edge is right-clicked, `subject` will be `IEdge`
* Canvas is right-clicked, so no node and no edge, `subject` will be `undefined`.

This is how `MOUSE_CLICK` works with `NODE_CLICK` and `EDGE_CLICK`. You can listen to `MOUSE_CLICK` and handle all three cases there if you want. This also enables someone to make a custom radius selection on top of the event position.